compiler/rustc_typeck/src/coherence/orphan.rs

```diff
@@ -277,42 +277,78 @@ fn emit_orphan_check_error<'tcx>(
         }
         traits::OrphanCheckErr::UncoveredTy(param_ty, local_type) => {
             let mut sp = sp;
-            for param in generics.params {
-                if param.name.ident().to_string() == param_ty.to_string() {
-                    sp = param.span;
+
+            match *param_ty.kind() {
+                ty::Projection(..) => {
+                    sp = self_ty_span;
```

Comment:
this is incorrect for things like
```rust
extern crate coherence_projection_assoc as lib;
use lib::Foreign;

trait Id {
    type Assoc;
}

impl<T> Id for T {
    type Assoc = T;
}

pub struct Local<T>(T);
impl<T> Foreign<<T as Id>::Assoc, Local<T>> for () {
    type Assoc = usize;
}

fn main() {}
```

Reply:
Hmm I'm going to need some guidance on how locate the type projection there... would I need to pass in more info into `emit_orphan_check_error`?

Reply:
What we probably should do is change the `OrphanChecker` to always know the index of the currently visited argument and return that. we could reuse that for type parameters as well as i think the new span would be more relevant.